### PR TITLE
Add Minecraft potion recipes and function for getting recipe chains

### PR DESCRIPTION
**Pull request by AbbeVass**
Will resolve this issue: https://github.com/JFarNTIG/crafterlib/issues/11

- [x]  My code follows the style guide (docs/style.md)
- [x]  Unit tests were added
- [ ]  Unit tests will be added later after some review
- [ ]  Unit tests weren't necessary

I added new recipes for a lot of Minecraft potions, their ingredients and new test data.

I also created a new craftutils function named `get_recipe_chain`. This function returns a path between two items in the form of a list of recipes. These recipes only includes ingredients and products. I chose to remove the rest of the data that's attached to each recipe because it wasn't needed and not including it made testing easier.

I added a number of new unit tests to test this function with different data, parameters and edge cases. All these tests are currently passing.

Lastly I added a new example that shows how this function works and how it can be used to display Minecraft potion recipes.

### CODE REVIEW

File: src/crafterlib/craftutils/recipe_chain.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    # Loop through every product in the path and collect their recipes
+    for i, item in enumerate(path[1:]):
+        recipes = game_data.get_recipes_for_item(item)
+        for recipe in recipes:
+            recipe = recipe.to_dict()
+
+            # Find the recipe with the previous product as an ingredient
+            # and remove unwanted data.
+            if path[i] in recipe['ingredients'].keys():
+                recipe.pop("id")
```
Comment (JFarNTIG) [inline src/crafterlib/craftutils/recipe_chain.py:84]: I understand what you're doing here, but this code is not very forward-compatible. As an example, imagine if a new field is added to recipe data at a later point. Your code will then not work as advertised, since recipes returned from this function will come with ingredients, products, and the new data field.

I would rewrite this so that you make a dict and specifically add only products and ingredients, rather than using the existing dict and specifically removing everything else.

**Review by JFarNTIG — CHANGES_REQUESTED** [review]
Great work, this is a nice and powerful feature for crafterlib to have.

I just had one concern, please read and make a revision and then we'll look at merging.

File: src/crafterlib/craftutils/recipe_chain.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+SPDX-License-Identifier: MIT
+"""
+import networkx as nx
+from crafterlib import GameCraftingData
+
+def get_recipe_chain(game_data: GameCraftingData,
+                     ingredient: str,
+                     product: str,
+                     combine_ingredients: bool = False) -> list:
```
Comment (JFarNTIG) [inline src/crafterlib/craftutils/recipe_chain.py:11]: The rest of the code uses `typing.List` as a type hint for lists, instead of `list`. I would recommend sticking to that convention. You can write:

`List[Dict]`

to refer to a list of dicts.

Note that you'll have to import these from `typing` to use them as type hints.

**Review by JFarNTIG — CHANGES_REQUESTED** [review]
